### PR TITLE
Fix error in type cast.

### PR DESCRIPTION
**Pull request by chingjun**
Casting to `List<String>` is an error when the original list is `List<dynamic>`.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/lib/src/device.dart [hunk 1]
```diff
@@ -993,7 +993,7 @@ class DebuggingOptions {
       debuggingEnabled: (json['debuggingEnabled'] as bool?)!,
       startPaused: (json['startPaused'] as bool?)!,
       dartFlags: (json['dartFlags'] as String?)!,
-      dartEntrypointArgs: ((json['dartEntrypointArgs'] as List<String>?)?.cast<String>())!,
+      dartEntrypointArgs: ((json['dartEntrypointArgs'] as List<dynamic>?)?.cast<String>())!,
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/device.dart:996]: I'm a little confused as to this fix:

> Casting to List is an error when the original list is List.

even after this change, the object could still be a list. It doesn't seem like calling `list.cast<T>()` where list is already a `List<T>` does not seem to be an error. Also, the test you added still passes without this fix.

Comment (chingjun) [reply packages/flutter_tools/lib/src/device.dart:996]: Github removed everything between `<>` in my original comment, I meant to say "casting to `List<String>` is an error when the original list is `List<dynamic>`."

Updated the original description.

And I missed a step in the test, I've now updated to convert it to JSON string, and parse the JSON into a Map<String, Object>.

The object returned from `json.decode` is a `List<dynamic>`, and it is an error to cast that to `List<String>` directly.

We should cast by using `(list as List<dynamic>).cast<String>()`

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/device.dart:996]: Ahh, ok, that makes sense now :)

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**